Citizen unit tests

mayx17 opened the pull request:
Unit tests for the new features and fixed som old tests.
eftersom att denna pr ser lite strulig ut är här en lista på det som gjorts:

-  allt i party integration package
- citizenResource: getPersonIdByPersonalNumber ändrades till att inkludera kommun id, rad 96-101
- CitizenService: getPersonIdByPersonalNumber ändrades till att gå mot party om kommun id 2281. rad 84-96
- tests: test för party integration, tests för de ändringar som gjorts i resource och service, fixade test som inte fungerade
- openapi-generator

mayx17 committed:
Integration + formating

Simon-n-o committed:
Small fixes to integration

mayx17 committed:
Integration update

adden some extra conifgurations

Simon-n-o committed:
openapi-generator added

Simon-n-o committed:
Integration fixed + formatting

mayx17 committed:
minor fix

changed log info too make it clearer

mayx17 committed:
fixed old tests

mayx17 committed:
Added new tests

added test for partyintegration

mayx17 committed:
Update CitizenAddressEntityTest.java

mayx17 committed:
fixed coverage for addressRepository

Simon-n-o committed:
small fix

mayx17 committed:
removed comments

removed unneeded comments

mayx17 committed:
formatting

martinhansson80 commented on pom.xml:
```diff
@@ -24,6 +28,11 @@
+		<dependency>
+			<groupId>se.sundsvall.dept44</groupId>
+			<artifactId>dept44-starter-feign</artifactId>
+			<version>6.0.10</version>
```
Versionen sätts automatiskt till samma som dept44, dvs den här bör tas bort

martinhansson80 commented on pom.xml:
```diff
@@ -40,12 +49,75 @@
 			<groupId>org.testcontainers</groupId>
 			<artifactId>mariadb</artifactId>
-			<scope>test</scope>
+			<version>1.19.1</version>
```
Gör en property av versionen, typ `<org.testcontainers.mariadb-version>1.19.1</org...>`

martinhansson80 commented on pom.xml:
```diff
@@ -40,12 +49,75 @@
+		<dependency>
+			<groupId>org.junit.jupiter</groupId>
+			<artifactId>junit-jupiter</artifactId>
+			<version>5.10.2</version>
```
Samma här, med versionen :)

martinhansson80 commented on src/main/java/se/sundsvall/citizen/integration/party/PartyClient.java:
```diff
@@ -0,0 +1,23 @@
+public interface PartyClient {
+	@GetMapping(value = "/{municipalityId}/{type}/{legalId}/partyId", produces = MediaType.TEXT_PLAIN_VALUE)
+
+	String getPartyId(@PathVariable("legalId") String personNumber,
```
Tips är att ändra metoden till att returnera en `<Optional>String` samt att sätta `dismiss404 = true` i `@FeignClient`.
`dismiss404 = true` gör att det inte kastas något exception vid 404 från Party.
Det blir tydligare i koden att det faktiskt kan saknas ett värde istället för att hantera null.

martinhansson80 commented on src/main/java/se/sundsvall/citizen/integration/party/configuration/PartyIntegrationProperties.java:
```diff
@@ -0,0 +1,19 @@
+import org.springframework.context.annotation.Primary;
+import se.sundsvall.citizen.integration.Oauth2;
+
+@Primary
```
Behövs `@Primary`?

martinhansson80 commented on src/main/java/se/sundsvall/citizen/service/CitizenService.java:
```diff
@@ -12,121 +23,118 @@
+		if (!Objects.equals(municipalityId, "2281")) {
+			final var citizenEntity = citizenRepository.findByPersonalNumber(personNumber)
+				.orElseThrow(() -> Problem.valueOf(NOT_FOUND,
+					format(ERROR_PERSONAL_NUMBER_NOT_FOUND, personNumber)));
```
Kan vara bra att inte exponera känslig information i felmeddelanden då det kommer tillbaka till klienten, typ personnummer :)

martinhansson80 commented on src/test/java/se/sundsvall/citizen/integration/db/model/CitizenAddressEntityTest.java:
```diff
@@ -14,102 +12,119 @@
-                .withCreatedAt(createdAt)
-                .withUpdatedAt(updatedAt);
+	@Test
+	void testEqualsAndHashCode() {
```
I think you're already testing hashCode using `hasValidBeanHashCode()`?
You could add `hasValidBeanEquals()` to that test as well.

mayx17 committed:
partyclient + minor fixes

changed return type to optional in party client, updated pom and fixed some tests

mayx17 committed:
Update README.md

added instructions for dependencies
